src/sql/workbench/contrib/editData/browser/editDataGridPanel.ts

```diff
@@ -1126,9 +1130,37 @@ export class EditDataGridPanel extends GridParentComponent {
 		});
 	}
 
-	onBeforeEditCell(event: Slick.OnBeforeEditCellEventArgs<any>): void {
+	onBeforeEditCell(event: Slick.OnBeforeEditCellEventArgs<any>): boolean {
+		let result = true;
 		this.logService.debug('onBeforeEditCell called with grid: ' + event.grid + ' row: ' + event.row
 			+ ' cell: ' + event.cell + ' item: ' + event.item + ' column: ' + event.column);
+
+		let itemToEdit = event.item[event.cell];
```

Comment:
please add comment explaining the scenarios when the itemToEdit could be a dbcellvalue or string or even an object with `text` property.

Reply:
Will do so, and even in the column formatter as well.